src/ansi-c/c_typecheck_code.cpp

```diff
@@ -828,9 +828,11 @@ void c_typecheck_baset::typecheck_spec_decreases(codet &code)
 {
   if(code.find(ID_C_spec_decreases).is_not_nil())
   {
-    exprt &variant = static_cast<exprt &>(code.add(ID_C_spec_decreases));
-
-    typecheck_expr(variant);
-    implicit_typecast_arithmetic(variant);
+    for(auto &decreases_clause_component :
+        (static_cast<exprt &>(code.add(ID_C_spec_decreases)).operands()))
```

Comment:
Are there any guarantees that this cast is safe? Take a look at this comment https://github.com/diffblue/cbmc/pull/5403#discussion_r468825167.

Reply:
Good catch. As suggested on the other thread, we should create "safe" accessors at some point.

However, currently, this issue also applies to invariant contracts, so may be we can table this change for a separate future PR and fix both together?

Reply:
Thanks for the link. For now, I will just leave it as it is. When someone cleans up the code for loop invariants, then we will also clean the code for decreases clauses.